### DISCUSSION TITLE
Is it possible to access the authenticated user in the integrated node server version?

### DISCUSSION THREAD

**Discussion by Ca1vino** [TiddlyWiki/TiddlyWiki5#5605] [Q&A]
If two different users have been authenticated into the server from separate browsers, can I access the authenticated user in WikiText coding? 

I'd like to provide some profile-dependent features.

**Comment by Jermolene** [comment]
Hi @Ca1vino

In the client-server configuration, the tiddler $:/status/UserName is set to the authenticated username.

You might also find the following useful, they all contain "yes" if they are true:

* $:/status/IsLoggedIn
* $:/status/IsAnonymous
* $:/status/IsReadOnly

**Reply by Ca1vino** [reply]
This is useful. I get the user name okay.  And some custom activity on that basis.

ButI'm getting hung up in the parsing, getting the name included into other titles, etc.  Specifically, putting something into $:/DefaultTiddlers that depends on $:/status/UserName.  It keeps coming through uninterpreted.

TW still confuses me about how to write code to calculate intermediate text snippets. Perhaps I need to use WikifyWidget as a kind of interpret-and-concatenate to a variable?

**Reply by Jermolene** [reply]
> ButI'm getting hung up in the parsing, getting the name included into other titles, etc. Specifically, putting something into $:/DefaultTiddlers that depends on $:/status/UserName. It keeps coming through uninterpreted.

To open a tiddler whose title includes the username you could put something like this in $:/DefaultTiddlers:

```
[{$:/status/UserName}addprefix[$:/state/userdata/]]
```

Thus, the user "Jermolene" would be presented with the tiddler "$:/state/userdata/Jermolene"

**Reply by Ca1vino** [reply]
Thanks, that helps, and I'm a couple steps closer to syntax enlightenment.

**Reply by Ca1vino** [reply]
But not that close, I guess. I don't understand why the two lines starting with "Set" don't produce the same result, so I have to recompute the filtered transclusion code rather than using the variable.

<$vars user-storylist-name="{{{ [{$:/status/UserName}addprefix[$:/state/userdata/]addsuffix[/StoryList]] }}}" 
main-storylist={{$:/StoryList!!list}}>
<$button>
Set <$text text={{{ [{$:/status/UserName}addprefix[$:/state/userdata/]addsuffix[/StoryList]] }}} /><br>
Set <$text text=<<user-storylist-name>> /> <br>
<$action-setfield $tiddler={{{ [{$:/status/UserName}addprefix[$:/state/userdata/]addsuffix[/StoryList]] }}} $field=list $value=<<main-storylist>> />
</$button>
</$vars>

**Reply by Ca1vino** [reply]
But getting back to the main question, the code above creates a user-dependent copy of the StoryList when user clicks a button, but what I actually want is for it to be automatic, so when a user relogs, their previous list is restored. It would need to be captured at some point (not sure where / how), then restored as a startup action.

**Comment by pmario** [comment]
@Jermolene 

I think that the 

>If **two different users** have been authenticated into the server from separate browsers

are the important question here. So can user A see, that user B is logged in and how?

**Reply by joshuafontany** [reply]
No, no user information is stored server-side. The user's credentials are sent as a cookie with every web request & each request is authenticated separately. There is currently no concept of a "user session" or "being online". I am working on a re-write to Bob that will add this type of "awareness".

**Reply by Ca1vino** [reply]
Bob is an impressive piece of work. I've got it running, but it doesn't update system tiddlers like StoryList, so I haven't been able to restore my set of working tiddlers between browser starts.

Do you have a guess about what's amiss?  See also -

https://github.com/Jermolene/TiddlyWiki5/discussions/5609
